extensions/EXT/GLX_EXT_no_config_context.txt

```diff
@@ -0,0 +1,254 @@
+    later versions of OpenGL ES, and OpenGL 3.0 and later versions of OpenGL.
+    Support for other versions and other client APIs is implementation
+    dependent. Such contexts are compatible with any drawable on the same
+    screen, though the API may impose further restrictions. For example,
+    OpenGL 3.0 contexts can not render to color index surfaces, even if color
+    index GLXFBConfigs are available."
+
+    In section 3.3.7 "Rendering Contexts" in the discussion of glXQueryContext,
+    to the paragraph beginning "The values and types", append:
+
+    "If the context was created without reference to a config, the value
+    of GLX_RENDER_TYPE will be GLX_DONT_CARE, and the values of
+    GLX_FBCONFIG_ID and GLX_VISUAL_ID_EXT will be GLX_NO_CONFIG_EXT."
+
+Interactions with GLX_ARB_fbconfig_float
+
+    A configuration created without reference to a config may be used to
+    render to surfaces of GLX_RGBA_FLOAT_TYPE_ARB in addition to RGBA
+    and color-index surfaces.
+
+Interactions with GLX_EXT_fbconfig_packed_float
+
+    A configuration created without reference to a config may be used to
+    render to surfaces of GLX_RGBA_UNSIGNED_FLOAT_TYPE_EXT in addition to
+    RGBA and color-index surfaces.
+
+Errors
+
+    No new error types are added.
+
+GLX Protocol
+
+    No new requests are added.
+
+    For GLXCreateContextAttribsARB, normally the client library derives the
+    <screen> field in the request from the specified fbconfig. When
+    GLX_NO_CONFIG_EXT is given as the fbconfig, the GLX_SCREEN attribute
+    and the <screen> field must match, or BadValue is generated. This implies
+    that the client library is responsible for pulling the screen number out
+    of the attribute list, since glXCreateContextAttribsARB does not have an
+    explicit screen parameter.
+
+Issues
+
+ 1) Should non-conformant no-config contexts be allowed to be created?
+
+    RESOLVED: No. We are not encouraging non-conformant contexts.
+
+ 2) Are no-config contexts constrained to those GL & ES implementations
+    which can support them?
+
+    RESOLVED: Yes. ES2 + OES_surfaceless_context, ES 3.0, and GL 3.0 all
+    support binding a surface without a context. This implies that they
+    don't need to know surface attributes at context creation time.
+
+ 3) For an OpenGL or OpenGL ES context created with no config, what is the
+    initial state of GL_DRAW_BUFFER and GL_READ_BUFFER for the default
+    framebuffer?
+
+    RESOLVED: This is an implementation detail rather than a spec issue.
+    glReadBuffer/glDrawBuffer have undefined results if called without a
+    current context. The GL_DRAW_BUFFER and GL_READ_BUFFER are set on the
+    first glXMakeCurrent call and can be updated in glReadBuffer and
+    glDrawBuffers calls after that. Therefore, the attribute value with
+    which the context is created is irrelevant from the point of view of the
+    spec and is left up to the implementation.
```

Comment:
I had a hell of a time getting this right when our implementation add no_surface support.  It's definitely a spec issue, I just don't think it's an issue for this spec to tackle.  Otherwise, what you've said all makes sense.  The state of the GL portion of a GLX context before its been made current is generally akin to the "If a tree falls in the woods..." thought exercise.  It only becomes interesting when some jerk calls glXCopyContext() on your context before it has been made current IIRC.